app/validators/certification_validator.rb

```diff
@@ -1,12 +1,16 @@
 class CertificationValidator < BaseValidator
   class << self
     def mandatory_fields
-      %i[certified_by certification_date]
+      %i[certification_type_id certified_by certification_date]
     end
   end
 
   private
 
+  def validate_certification_type_id
+    validate_presence(:certification_type_id, :blank) if @record&.claim&.agfs?
```

Comment:
To match current logic  this needs to be 

```suggestion
    validate_presence(:certification_type_id, :blank) if @record&.claim&.agfs? && @record&.claim&.final?
```
or 
```suggestion
    validate_presence(:certification_type_id, :blank) if @record&.claim.is_a?(Claim::AdvocateClaim)
```

i prefer the less specific nature of the first myself, even if a bit messy

Reply:
OR
```suggestion
  return unless if @record&.claim&.agfs? && @record&.claim&.final?
  validate_presence(:certification_type_id, :blank) 
```

Reply:
have applied commit for this